ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -397,13 +426,17 @@ long getOpenSegmentSizeInBytes() {
     return openSegment == null ? 0 : openSegment.getTotalFileSize();
   }
 
-  private long getTotalCacheSize() {
-    return closedSegments.getTotalCacheSize() +
-            Optional.ofNullable(openSegment).map(LogSegment::getTotalCacheSize).orElse(0L);
+  private long getTotalCacheSize(LogSegmentListCacheInfo
+                                     closedSegmentsCacheInfo) {
+    return closedSegmentsCacheInfo.getTotalCacheSize() +
+        Optional.ofNullable(openSegment).map(LogSegment::getTotalCacheSize).orElse(0L);
```

Comment:
Let's combine this method with `shouldEvict()`.

Reply:
Updated according to the suggested patch diff.